engine/backends/proverif/proverif_backend.ml

```diff
@@ -419,6 +439,16 @@ module Print = struct
           ^^ space ^^ string "in" ^^ hardline
           ^^ (print#expr_at Expr_Let_body body |> group)
 
+      method concrete_ident' ~(under_current_ns : bool) : concrete_ident fn =
+        fun id ->
+          if under_current_ns then print#name_of_concrete_ident id
+          else
+            let crate, path = print#namespace_of_concrete_ident id in
+            let full_path = crate :: path in
+            separate_map (underscore ^^ underscore) utf8string full_path
+            ^^ underscore ^^ underscore
```

Comment:
we might want to move this renaming to the name policy thing at some point.
Otherwise, name collisions are possible
But for now that's completely fine, let's see that later

Reply:
For sure, if we had like a `path_separator` that you could specify in the name policy, then this could be a bit simpler. :+1: 
Do you mean collisions are possible in general with this ad hoc method of defining a path renaming scheme, or with this specific one? I thought I could avoid collisions here by including the full path always.

Reply:
Yes, exactly, I think we should do that at some point. I'm basically doing something similar, but only in certain cases (e.g. I collapse certain kind of namespaces, not all in all circumstances)

What I mean is that Rust identifiers `foo::a__b` and `foo::a::b` will collide with the current renaming scheme.
It's unlikely, but that'd be nice to avoid. In the name policy thing, I escape all separators to avoid that, but that's a bit cumbersome, so yeah, for now I think we don't care, and then we'll move that logic in the name policy module at some point :)

Reply:
Oops, I forgot about unescaped separators! Thanks for pointing this out.